README.md

```diff
@@ -8,12 +8,13 @@
 ### Warning!!!: This package is still work-in-progress
 
 User API:
+`D(f)(x)` returns a lazy representation of the derivative. 
 
-`D(f)(x) * v` computes `df(x)/dx ⋅ v`.
+`D(f)(x) * v` computes `df(x)/dx ⋅ v`, taking advantage of the laziness in `D(f)(x)`. 
 
 
-`DI(f)(x)` is a convenient function to compute the derivative, gradient or
-Jacobian of `f` at `x`.
+`DI(f)(x)` is a convenience function to materialize the derivative, gradient or
+Jacobian of `f` at `x`. Equivalent to D(f)(x) * I
```

Comment:
> Equivalent to D(f)(x) * I

Not really. For instance, `D(sin)(1) * I` is wrong. It is better to not mention it. Talking about multiplicative identity is also not really helpful here. Less is more here. The user doesn't need more info than "it materializes the computation".